src/sentry/notifications/utils/__init__.py

```diff
@@ -268,9 +268,6 @@ def get_interface_list(event: Event) -> Sequence[tuple[str, str, str]]:
 
 
 def send_activity_notification(notification: ActivityNotification | UserReportNotification) -> None:
-    if not notification.should_email():
```

Comment:
@mgaeta Definitely outside the scope this PR, but can we get rid of `send_activity_notification` and instead just override the implementation of class methods?

Reply:
The problem is that `send_activity_notification()` is used by every `ActivityNotification` but also by `UserReportNotification`, which isn't a `ActivityNotification`. We could get rid of this helper by making an `Activity` for user reports and making `UserReportNotification` an `ActivityNotification`.